### PR TITLE
Huge refactor

### PR DESCRIPTION
**Pull request by fiestacasey**
Added specific exceptions and error checking to each provider
Refactored regex and tracking_url into providers
Moved some stuff around to fit standard practices better. (renamed OmniShip to Omniship for best practices too. It was either that or renaming the files from omniship to omni_ship)
Wrote tests!

@ksolkowski @kahlya

### CODE REVIEW

**Comment by fiestacasey** [comment]
@ksolkowski @kahlya  ?

**Comment by austinkahly** [comment]
This looks way nicer, and the testing is super nice. 👍

File: lib/omniship/dhlgm.rb [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+
     class << self
       attr_accessor :username
       attr_accessor :password
       attr_accessor :mailer_id
     end
 
+
+    def self.tracking_test?(tracking)
+      !Omniship::DHLGM.mailer_id.nil? and tracking.length == TRACKING_LENGTH and tracking.include?(Omniship::DHLGM.mailer_id)
```
Comment (ksolkowski) [inline lib/omniship/dhlgm.rb:17]: `Omniship::DHLGM.mailer_id.present?` is easier to read then `!.nil?`

File: lib/omniship/usps.rb [hunk 2]
```diff
@@ -23,28 +29,36 @@ class << self
...
-		    :city => pdu_city,
-		    :state => pdu_state,
-		    :zip5 => pdu_zip5
-		}
+      retailer = {
+        :name => retailer_name,
+        :address => retailer_address
+      }
+      permit = {
+        :number => permit_number,
```
Comment (ksolkowski) [inline lib/omniship/usps.rb]: no rockets plz

File: lib/omniship/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
-module OmniShip
-  VERSION = '0.1.28'
+module Omniship
+  VERSION = '0.2'
```
Comment (ksolkowski) [inline lib/omniship/version.rb:2]: 👏

**Comment by ksolkowski** [comment]
fancy.jpg 👍 🐫

File: lib/omniship/dhlgm.rb (hunk 1 above)
Comment (fiestacasey) [inline lib/omniship/dhlgm.rb:17]: present? is a rails function. We're outside of rails here

Comment (ksolkowski) [inline lib/omniship/dhlgm.rb:17]: oh.... TIL

**Pull request merged**